relay-server/src/services/project_redis.rs

```diff
@@ -52,19 +52,44 @@ impl RedisProjectSource {
         RedisProjectSource { config, redis }
     }
 
-    pub fn get_config(&self, key: ProjectKey) -> Result<ProjectState, RedisProjectError> {
-        let mut command = relay_redis::redis::cmd("GET");
-
-        let prefix = self.config.projectconfig_cache_prefix();
-        command.arg(format!("{prefix}:{key}"));
+    pub fn get_config_if_changed(
+        &self,
+        key: ProjectKey,
+        revision: Option<&str>,
+    ) -> Result<Option<ProjectState>, RedisProjectError> {
```

Comment:
Instead of `Result<Option>` I would rather return here an Enum which would better signal what the actual result of the `get` is.  I think it would be nicer to read.

Reply:
Replacing the entire `Result<Option<>>` with an enum makes error handling really verbose, since `?` can no longer be used.

Replacing the `Option<>` with an enum is pretty verbose imo, I added some documentation describing the behaviour to make it more clear instead!